src/CommandParser.h

```diff
@@ -38,67 +39,89 @@ struct StopCommandInfo {
   std::string ServiceID;
 };
 
-class CommandParser {
-public:
-  /// \brief Extract the information from the start command.
-  ///
-  /// \param JSONCommand The JSON Command.
-  /// \param DefaultStartTime The start time to use if not supplied in the JSON
-  /// \return The start information.
-  StartCommandInfo extractStartInformation(
-      const nlohmann::json &JSONCommand,
-      std::chrono::milliseconds DefaultStartTime = getCurrentTime());
-
-  /// \brief Extract the information from the stop command.
-  ///
-  /// \param JSONCommand The JSON Command.
-  /// \return The stop information.
-  static StopCommandInfo
-  extractStopInformation(const nlohmann::json &JSONCommand);
-
-  /// \brief Extract the command name from the command.
-  ///
-  /// Note: the command is converted to lower-case.
-  ///
-  /// \param JSONCommand The JSON Command.
-  /// \return The command name.
-  static std::string extractCommandName(const nlohmann::json &JSONCommand);
-
-  static std::string const StopCommand;
-  static std::string const StartCommand;
-  static std::string const ExitCommand;
-  static std::string const StopAllWritingCommand;
-
-private:
-  SharedLogger Logger = getLogger();
-
-  uri::URI extractBroker(nlohmann::json const &JSONCommand);
-  static std::string extractJobID(nlohmann::json const &JSONCommand);
-  static std::chrono::duration<long long int, std::milli> getCurrentTime();
-  static std::chrono::milliseconds
-  extractTime(std::string const &Key, nlohmann::json const &JSONCommand,
-              std::chrono::milliseconds const &DefaultTime);
-
-  template <typename T>
-  static T getRequiredValue(std::string const &Key,
-                            nlohmann::json const &JSONCommand) {
-    if (auto x = find<T>(Key, JSONCommand)) {
-      return x.inner();
-    }
-
-    throw std::runtime_error(
-        fmt::format("Missing key {} from command JSON", Key));
+namespace CommandParser {
+static std::string const StartCommand = "filewriter_new";
```

Comment:
It is possible to hide these string constants in the *.cpp-file. Of course, this means that they are unavailable for (e.g.) unit testing. Thus, I am ok with leaving them as is.

Reply:
They are needed in `CommandHandler` I think.

Reply:
Ah, ok.